sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLogEmitter.java

```diff
@@ -29,21 +28,28 @@
 
   @Override
   public LogBuilder logBuilder() {
-    return new SdkLogBuilder();
+    return new SdkLogBuilder(logEmitterSharedState, instrumentationLibraryInfo);
   }
 
   // VisibleForTesting
   InstrumentationLibraryInfo getInstrumentationLibraryInfo() {
     return instrumentationLibraryInfo;
   }
 
-  private final class SdkLogBuilder implements LogBuilder {
```

Comment:
Besides some extra boilerplate 😁, what do we gain from this?

Reply:
In my brain, the implicit sharing of state (instance fields) between an outer and nonstatic inner class is much harder to reason about than just a vanilla class. Alternatively, we could make it a pure anonymous class or pull it up to the top level I suppose....but this seemed like a good compromise.

Reply:
Slight preference for making a top level class, just to be consistent with span

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java

Reply:
I like consistency, and think its fine to make this a top level class since its package private so ultimately an implementation detail.

In general though private classes (both static and non-static) are beneficial in that they make it clear to maintainers that class is only intended to be used by the outer class. I don't think we should make it a rule in this project that all classes should be top level, or that inner non-static classes aren't allowed - these language features have their place.

Reply:
Ok, thanks for this feedback. I'll move it up for consistency. 👍🏻